File: shared/src/main/java/com/screentime/shared/model/UsageSnapshot.kt
package com.screentime.shared.model

data class UsageSnapshot(
    val date: String,
    val perAppMillis: Map<String, Long>,
) {
    fun millisFor(packageName: String): Long = perAppMillis[packageName] ?: 0L

    fun totalMillis(): Long = perAppMillis.values.sum()
}

File: web/src/models/UsageSnapshot.ts
export interface UsageSnapshot {
  date: string;
  perAppMillis: Record<string, number>;
}

export function millisFor(snapshot: UsageSnapshot, packageName: string): number {
  return snapshot.perAppMillis[packageName] ?? 0;
}

export function totalMillis(snapshot: UsageSnapshot): number {
  return Object.values(snapshot.perAppMillis).reduce((sum, v) => sum + v, 0);
}
